auction/src/lib.rs

```diff
@@ -180,16 +182,21 @@ impl<T: Trait> Auction<T::AccountId, T::BlockNumber> for Module<T> {
 		Ok(())
 	}
 
-	fn new_auction(start: T::BlockNumber, end: Option<T::BlockNumber>) -> Self::AuctionId {
+	fn new_auction(start: T::BlockNumber, end: Option<T::BlockNumber>) -> result::Result<Self::AuctionId, DispatchError> {
 		let auction = AuctionInfo { bid: None, start, end };
 		let auction_id = Self::auctions_index();
-		<AuctionsIndex<T>>::mutate(|n| *n += Self::AuctionId::one());
+		<AuctionsIndex<T>>::try_mutate(|n| -> DispatchResult {
```

Comment:
```suggestion
		let auction_id = <AuctionsIndex<T>>::try_mutate(|n| -> DispatchResult {
```

and remove line above and make it return `Result<AucionIndex, DispatchError>`